sklearn/random_projection.py

```diff
@@ -128,7 +128,7 @@ def johnson_lindenstrauss_min_dim(n_samples, eps=0.1):
             % n_samples)
 
     denominator = (eps ** 2 / 2) - (eps ** 3 / 3)
-    return (4 * np.log(n_samples) / denominator).astype(np.int)
+    return (4 * np.log(n_samples) / denominator).astype(np.float64)
```

Comment:
Returning a float would break backward compatibility and would not be consistent with the function docstring,
```
    Returns
    -------
    n_components : int or numpy array of int,
        The minimal number of components to guarantee with good probability
        an eps-embedding with n_samples.

```

We can return an int64 though I think.